pincer/objects/guild/guild.py

```diff
@@ -1916,6 +1919,266 @@ async def get_webhooks(self) -> AsyncGenerator[Webhook, None]:
+
+    async def modify_scheduled_event(
+        self,
+        _id: int,
+        name: Optional[str] = None,
+        entity_type: Optional[int] = None,
+        privacy_level: Optional[int] = None,
+        scheduled_start_time: Optional[datetime] = None,
+        scheduled_end_time: Optional[datetime] = None,
+        entity_metadata: Optional[str] = None,
+        channel_id: Optional[int] = None,
+        description: Optional[str] = None,
+        status: Optional[int] = None,
+        reason: Optional[str] = None,
+    ) -> ScheduledEvent:
+        """
+        Modify a scheduled event.
+
+        Parameters
+        ----------
+        _id : :class:`int`
+            The id of the scheduled event.
+        name : Optional[:class:`str`]
+            The name of the scheduled event.
+        entity_type : Optional[:class:`int`]
+            The type of entity to be scheduled.
+        privacy_level : Optional[:class:`int`]
+            The privacy level of the scheduled event.
+        scheduled_start_time : Optional[:class:`datetime`]
+            The scheduled start time of the event.
+        scheduled_end_time : Optional[:class:`datetime`]
+            The scheduled end time of the event.
+        entity_metadata : Optional[:class:`str`]
+            The metadata of the entity to be scheduled.
+        channel_id : Optional[:class:`int`]
+            The channel id of the channel to be scheduled.
+        description : Optional[:class:`str`]
+            The description of the scheduled event.
+        status : Optional[:class:`int`]
+            The status of the scheduled event.
+        reason : Optional[:class:`str`]
+            The reason for modifying the scheduled event.
+
+        Raises
+        ------
+        :class:`ValueError`
+            If the scheduled event is in the past,
+            or if the scheduled end time is before the scheduled start time.
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.scheduled_event.ScheduledEvent`
+            The scheduled event object.
+        """
+        if scheduled_start_time:
+            timestamp = scheduled_start_time.timestamp()
+            if timestamp < datetime.now().timestamp():
+                raise ValueError("An event cannot be created in the past")
+
+            if (
+                scheduled_end_time
+                and timestamp > scheduled_end_time.timestamp()
+            ):
+                raise ValueError("An event cannot start before it ends")
+
+        kwargs: Dict[str, str] = remove_none(
+            {
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat()
+                if scheduled_start_time is not None
+                else None,
+                "scheduled_end_time": scheduled_end_time.isoformat()
+                if scheduled_end_time is not None
+                else None,
```

Comment:
Maybe same here?
```suggestion
                "scheduled_start_time": getattr(
                    scheduled_start_time, "isoformat", type(None)
                )(),
                "scheduled_end_time": getattr(
                    scheduled_end_time, "isoformat", type(None)
                )(),
```

Reply:
I don't think I'll change this, I find that this change is not very readable

Reply:
Alrighty, resolve my comments then